src/SourceBuild/content/prep.sh

```diff
@@ -192,3 +197,14 @@ fi
 if [ "$downloadPrebuilts" == true ]; then
   DownloadArchive Prebuilts false $artifactsRid
 fi
+
+if [ "$keepBinaries" == false ]; then
+  git config --global --add safe.directory /vmr
+  
+  if git ls-files | grep -q -i '\.\(dll\|exe\|mdb\|nupkg|\pbd\|tgz\|zip\)$'; then
```

Comment:
Consider avoiding code duplication here as well as having to call ls-files multiple times.